src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -382,61 +379,6 @@ state switch
                 compilations = new CompilationPair(
                     compilationWithoutGeneratedDocuments,
                     compilationWithoutGeneratedDocuments.AddSyntaxTrees(generatorInfo.Documents.States.Values.Select(state => state.GetSyntaxTree(cancellationToken))));
-
-                // Now add in back a consistent set of project references.  For project references
-                // try to get either a CompilationReference or a SkeletonReference. This ensures
-                // that the in-progress project only reports a reference to another project if it
-                // could actually get a reference to that project's metadata.
-                var metadataReferences = new List<MetadataReference>();
-                var newProjectReferences = new List<ProjectReference>();
-                metadataReferences.AddRange(this.ProjectState.MetadataReferences);
-
-                metadataReferenceToProjectId = [];
-
-                foreach (var projectReference in this.ProjectState.ProjectReferences)
```

Comment:
@ToddGrun So this code simply attempts to ensure that the p2p refs in the frozen view of the world point at something.  because it's partial and we want to be fast, a p2p ref will point at the *compilation without generated docs* for the linked project.

previously, we did this work ehre.  However, because we're now just making ordinary (albeit frozen) inprogressstates, we can get rid of this.  Instead, when we then do the similar pass when transitioning an inprogressstate to a finalstate, we do this same work (deviating only slightly to preserve the isfrozen speedup).